### ISSUE TITLE
Keras model prediction results are not classified properly in confusion matrix

### ISSUE THREAD

**Issue by Wazaki-Ou** [keras-team/keras#11227]
I am fine tuning VGG16 to detect 4 classes on my dataset and while training and validation accuracy increases, when I finish training and I predict and use sklearn confusion_matrix and classification_report, it always gives me a 25% accuracy (even when I try to predict on training and validation sets). I tuned parameters and tried many solutions thinking my model was not learning properly, but then realized that when using model.evaluate() the accuracy is higher and makes more sense. I came down to the conclusion that there is maybe an issue with sklearn functions, but then I followed tutorials and documentation code so I'm not sure if there is a problem in my code or of compatibility issue between keras and sklearn in how they represent data. This is the part of the code where I use the two functions (tested on the dogs/cats dataset where the sklearn functions give 50% accuracy).

```
def plot_confusion_matrix_two(cm,
                          target_names,
                          title='Confusion matrix',
                          cmap=None,
                          normalize=True):

import matplotlib.pyplot as plt
import numpy as np
import itertools

accuracy = np.trace(cm) / float(np.sum(cm))
misclass = 1 - accuracy

if cmap is None:
    cmap = plt.get_cmap('Blues')

plt.figure(figsize=(8, 6))
plt.imshow(cm, interpolation='nearest', cmap=cmap)
plt.title(title)
plt.colorbar()

if target_names is not None:
    tick_marks = np.arange(len(target_names))
    plt.xticks(tick_marks, target_names, rotation=45)
    plt.yticks(tick_marks, target_names)

if normalize:
    cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]


thresh = cm.max() / 1.5 if normalize else cm.max() / 2
for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):
    if normalize:
        plt.text(j, i, "{:0.4f}".format(cm[i, j]),
                 horizontalalignment="center",
                 color="white" if cm[i, j] > thresh else "black")
    else:
        plt.text(j, i, "{:,}".format(cm[i, j]),
                 horizontalalignment="center",
                 color="white" if cm[i, j] > thresh else "black")


plt.tight_layout()
plt.ylabel('True label')
plt.xlabel('Predicted label\naccuracy={:0.4f}; misclass={:0.4f}'.format(accuracy, misclass))
plt.show()
```

```
history = model.fit_generator(
    train_generator,
    steps_per_epoch=train_generator.samples // train_generator.batch_size,
    epochs=epochs,
    validation_data=validation_generator,
    validation_steps=validation_generator.samples // validation_generator.batch_size,
    #callbacks=[ModelCheckpoint('VGG16-transferlearning.model', monitor='val_acc', save_best_only=True)]
    verbose=1
)
```

```
predictions = model.predict_generator(test_generator, steps=test_generator.samples//test_generator.batch_size, 
                                      verbose=0)

#Confution Matrix and Classification Report
predictions = np.argmax(predictions, axis=1)

print('Confusion Matrix')
cm = confusion_matrix(test_generator.classes, predictions)
#print(cm)

target_names =['cats', 'dogs']
plot_confusion_matrix_two(cm, target_names, title='Confusion Matrix',cmap=None,normalize=False)

print('Classification Report')
print(classification_report(test_generator.classes, predictions, target_names=target_names))

print('Confusion Matrix')
print(cm)
```

Am I feeding the wrong input to the functions ?? the data_generators have the attribute shuffle set as True by the way, but when print test_generator.classes it returns an array of labels in order.

I'll be very thankful if someone can help me solve this issue or suggest a different library to use. I would like to get a good representation of the prediction.
